org.adempiere.base/src/org/compiere/model/MInOutLineMA.java

```diff
@@ -277,6 +277,15 @@ protected boolean beforeSave (boolean newRecord)
 				dateMPolicy = inout.getMovementDate();
 			}
 			
+			setDateMaterialPolicy(dateMPolicy);
+		} else if (newRecord && getM_AttributeSetInstance_ID() > 0 && getDateMaterialPolicy() == null) {
+			I_M_InOutLine line = getM_InOutLine();

```

Comment:
Hi @hengsin - I see this is basically copied from the lines above, two considerations:

1 - wouldn't be easier and better for maintenance to add an OR to the condition in line 264
2 - the getM_InOutLine() and line.getM_InOut() are really bad for performance - calling MTable.get(...).getPO(...) - I think for performance would be better to avoid that in core (or improve the generated methods to be faster)